### PR TITLE
[READY] - nix: refactor and update

### PR DESCRIPTION
**Pull request by Kranzes**
## Description of PR

Relates to: https://github.com/socallinuxexpo/scale-network/issues/627

- Moving `perl` tests to be trigger via nix instead of from inside docker
- Refactoring nix to leverage [flake.parts](flake.parts) to utilize the module system for all aspects in the flake

### Update

Going to probably unify more of our gha tests into a single gha workflow.yml in a follow up.

## Previous Behavior
- Testing `perl` ci via docker
- Not using flake.parts within the flake

## New Behavior
- `perl` runs via nix checks in ci
- flake.parts and refactoring of the nix code for the flake

## Tests

- CI tests pass

Can run all of our tests in one go via:

```
nix flake check
```

### CODE REVIEW

**Comment by sarcasticadmin** [comment]
Thanks for raising this! Ill take this for a spin shortly!

**Review by sarcasticadmin — APPROVED** [review]
@Kranzes thanks a lot for all of the help getting this refactored into flake.parts, it looks awesome and seems really helpful for us going forward! :shipit: 

Ill wait for others to comment/review since I added a couple of fixups to the branch for the CI tests.

File: .github/workflows/validate-datafiles.yml
Comment (Kranzes) [inline .github/workflows/validate-datafiles.yml:19]: if you run `nix flake check` it will evaluate all the code in the repo, even nixos configurations and build all derivations under the `checks` output.

Comment (sarcasticadmin) [reply .github/workflows/validate-datafiles.yml:19]: agreed, but I think itll have an issue running all tests due to the need for qemu in github actions

**Review by owendelong — APPROVED** [review]
I don't see any issues, but honestly, I don't really know what I'm looking at, so my comments are of minimal value here.

**Comment by sarcasticadmin** [comment]
@owendelong I'm happy to give you a primer on more of this during the work party, thanks for looking at it regardless

**Comment by Kranzes** [comment]
Once we get this merged I can work with @sarcasticadmin on removing dead code and repeating ourselves a bit less by using some functions magic.

**Review by kylerisse — APPROVED** [review]
Thanks @Kranzes . Appreciate the help.

**Pull request merged**
